src/coreclr/jit/fgopt.cpp

```diff
@@ -5624,8 +5624,10 @@ bool Compiler::fgUpdateFlowGraph(bool doTailDuplication)
                                 BasicBlock* const bFixup = fgNewBBafter(BBJ_ALWAYS, bDest, true);
                                 bFixup->inheritWeight(bDestNext);
                                 bFixup->bbJumpDest = bDestNext;
-                                fgReplacePred(bDestNext, bDest, bFixup);
+
+                                fgRemoveRefPred(bDestNext, bDest);
                                 fgAddRefPred(bFixup, bDest);
+                                fgAddRefPred(bDestNext, bFixup);
                             }
```

Comment:
So ReplacePred should always be avoided and be replaced with a pair of RemoveRefPred + AddRefPred ?

Reply:
Not necessarily, sometimes you want to replace all the references.

I find editing flow in the jit to be fairly complex in general, especially dealing with the fall through behavior. I have ideas on how to simplify this but not sure when/if we'll get to them.